Upstream-Commit effadce6c756247ea8bae32dc13bb3e6f464f0eb (FFmpeg 6.1): Konstanten für
Schiebebefehle in Inline-Assembler, sonst bricht der Build mit binutils ab 2.41 ab.

--- a/libavcodec/x86/mathops.h	2020-04-27 23:48:15.000000000 +0200
+++ b/libavcodec/x86/mathops.h	2026-10-26 16:10:14.130947564 +0100
@@ -113,19 +113,31 @@
 // avoid +32 for shift optimization (gcc should do that ...)
 #define NEG_SSR32 NEG_SSR32
 static inline  int32_t NEG_SSR32( int32_t a, int8_t s){
-    __asm__ ("sarl %1, %0\n\t"
-         : "+r" (a)
-         : "ic" ((uint8_t)(-s))
-    );
+    if (__builtin_constant_p(s))
+        __asm__ ("sarl %1, %0\n\t"
+             : "+r" (a)
+             : "i" (-s & 0x1F)
+        );
+    else
+        __asm__ ("sarl %1, %0\n\t"
+             : "+r" (a)
+             : "c" ((uint8_t)(-s))
+        );
     return a;
 }
 
 #define NEG_USR32 NEG_USR32
 static inline uint32_t NEG_USR32(uint32_t a, int8_t s){
-    __asm__ ("shrl %1, %0\n\t"
-         : "+r" (a)
-         : "ic" ((uint8_t)(-s))
-    );
+    if (__builtin_constant_p(s))
+        __asm__ ("shrl %1, %0\n\t"
+             : "+r" (a)
+             : "i" (-s & 0x1F)
+        );
+    else
+        __asm__ ("shrl %1, %0\n\t"
+             : "+r" (a)
+             : "c" ((uint8_t)(-s))
+        );
     return a;
 }
 
